### PR TITLE
fix - issue 1080

### LINKED ISSUE #1080: Possible error in 4.9.2 Testing for Padding Oracle
**What's the issue?**
I think there is error on line 43 in the formula for calculating the number of blocks (*`b = ceil( length(b) / n)`*).
https://github.com/OWASP/wstg/blob/2534bbfee762a87d42c7f91d301d10b086bd420e/document/4-Web_Application_Security_Testing/09-Testing_for_Weak_Cryptography/02-Testing_for_Padding_Oracle.md?plain=1#L43C1-L43C1

**How do we solve it?**
I think that the right formula should be: *`b = ceil( length(p) / n)`*, where *p* is the plaintext.

Would you like to be assigned to this issue?
Check the box if you will submit a PR to fix this issue. Please read CONTRIBUTING.md.
- [x] Assign me, please!

### PR DESCRIPTION
**Pull request by Sacek073**
This PR fixes #1080. 

- [x] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- It solves the error in formula for computing number of blocks.

### CODE REVIEW

**Comment by Sacek073** [comment]
Just want to let you know, that these errors are not from my edit. Thank you

**Comment by kingthorin** [comment]
That's okay we can still do the right thing and go ahead and improve it.

File: document/4-Web_Application_Security_Testing/09-Testing_for_Weak_Cryptography/02-Testing_for_Padding_Oracle.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ First the possible input points for padding oracles must be identified. Generall
 
 `Dg6W8OiWMIdVokIDH15T/A==` results after Base64 decoding in `0e 0e 96 f0 e8 96 30 87 55 a2 42 03 1f 5e 53 fc`. This seems to be random and 16 byte long.
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/09-Testing_for_Weak_Cryptography/02-Testing_for_Padding_Oracle.md:41]:
```suggestion
`Dg6W8OiWMIdVokIDH15T/A==` results after base64 decoding in `0e 0e 96 f0 e8 96 30 87 55 a2 42 03 1f 5e 53 fc`. This seems to be random and 16 byte long.
```

File: document/4-Web_Application_Security_Testing/09-Testing_for_Weak_Cryptography/02-Testing_for_Padding_Oracle.md [hunk 2]
```diff
@@ -40,7 +40,7 @@ First the possible input points for padding oracles must be identified. Generall
...
+If such an input value candidate is identified, the behavior of the application to bit-wise tampering of the encrypted value should be verified. Normally this Base64 encoded value will include the initialization vector (IV) prepended to the cipher text. Given a plaintext *`p`* and a cipher with a block size *`n`*, the number of blocks will be *`b = ceil( length(p) / n)`*. The length of the encrypted string will be *`y=(b+1)*n`* due to the initialization vector. To verify the presence of the oracle, decode the string, flip the last bit of the second-to-last block *`b-1`* (the least significant bit of the byte at *`y-n-1`*), re-encode and send. Next, decode the original string, flip the last bit of the block *`b-2`* (the least significant bit of the byte at *`y-2*n-1`*), re-encode and send.
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/09-Testing_for_Weak_Cryptography/02-Testing_for_Padding_Oracle.md:43]:
```suggestion
If such an input value candidate is identified, the behavior of the application to bit-wise tampering of the encrypted value should be verified. Normally this base64 encoded value will include the initialization vector (IV) prepended to the cipher text. Given a plaintext *`p`* and a cipher with a block size *`n`*, the number of blocks will be *`b = ceil( length(p) / n)`*. The length of the encrypted string will be *`y=(b+1)*n`* due to the initialization vector. To verify the presence of the oracle, decode the string, flip the last bit of the second-to-last block *`b-1`* (the least significant bit of the byte at *`y-n-1`*), re-encode and send. Next, decode the original string, flip the last bit of the block *`b-2`* (the least significant bit of the byte at *`y-2*n-1`*), re-encode and send.
```

**Review by kingthorin — APPROVED** [review]

**Pull request merged**